src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,468 @@
 -----------------------------------------------------------------------------
 module Algebra.Graph.Relation.Symmetric (
     -- * Data structure
-    SymmetricRelation, fromRelation, toRelation,
+    SymmetricRelation(..), fromRelation, toRelation,
+
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Relations on graphs
+    isSubgraphOf,
 
     -- * Graph properties
-    neighbours
+    isEmpty, hasVertex, hasEdge, vertexCount, edgeCount, vertexList, edgeList,
+    adjacencyList, vertexSet, edgeSet, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices, gmap, induce,
+
+    -- * Relational operations
+    compose
   ) where
 
-import Algebra.Graph.Relation
-import Algebra.Graph.Relation.InternalDerived
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI hiding (empty, vertex, overlay, connect, referredToVertexSet)
+import Algebra.Graph.Relation.Symmetric.Internal
 
 import qualified Data.Set as Set
+import Data.Tree 
+import qualified Data.Tree as Tree
+import Data.Tuple (swap)
 
 -- | Construct a symmetric relation from a 'Relation'.
 -- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
+fromRelation :: R.Relation a -> SymmetricRelation a
 fromRelation = SymmetricRelation
 
 -- | Extract the underlying relation.
 -- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
```

Comment:
Could you give an example?

Reply:
An example as how would this function be useful?

Reply:
Yes. I aim to illustrate all functions in the API by some meaningful examples, which also typically serve as good tests.